hack/scripts/validate_state.sh

```diff
@@ -17,6 +17,9 @@ do
     echo "Node internal ip: $node_ip"
     privileged_pod=$(kubectl get pods -n kube-system -l app=privileged-daemonset -o wide | grep "$node_name" | awk '{print $1}')
     echo "privileged pod : $privileged_pod"
+    if [ "$privileged_pod" == '' ]; then
+        exit 1
```

Comment:
Did we encounter such case during testing ?

Can we add the status of privileged pod deployment then. ( `kubectl describe daemonset privileged-daemonset -n kube-system`

Reply:
sure I'll add that, here you can see this run just got stuck in the loop. I had to manually cancel it https://dev.azure.com/msazure/One/_build/results?buildId=71680459&view=logs&j=4ea62961-c456-50ab-e773-f15fbc744993&t=6637b73f-d7ef-5d5e-d4d4-eb0bbec757cb&s=c689f5d8-16f1-5a52-95fe-f6a4e6a9e7fe

Reply:
Sorry I meant if we fail to get the privileged pod then it will be good check the status of daemon set. Basically before exiting we can have that as I see from the pipeline run, `daemonset.apps/privileged-daemonset` created.

Ideally we should be waiting for the deployment to be complete before proceeding i think.